Adds the Dark Matt-eor when you emag a stupid amount of meteor shields + lots of meteor file sorting + qol + dark matter singularity + dark matt-eor summoning final traitor objective

tralezab opened the pull request:
## About The Pull Request

<details>
  <summary>Dark Matt-eor Image</summary>
  
  ![image](https://user-images.githubusercontent.com/40974010/228368755-34ae5f89-e1bb-498b-bbf8-a14ff4240dc0.png)

</details>

> A barely visible blur in the cosmic darkness, like a ghostly shadow on a moonless night. A piercing howl in the vacuum of space, as if it were tearing the fabric of reality. A twisted halo of light around it, bending and breaking the rays of distant suns. A shower of quantum sparks, flickering and fading in its wake. A dark matter meteor (dark matt-eor) is a wonder to witness, and to dread.

> A sudden impact, like a hammer blow to the heart of the station. A violent tremor, shaking and shattering the metal walls and windows. A deafening roar, as the air rushes out of the breached hull. A blinding flash, as the dark matter meteor unleashes its hidden energy. A tiny black hole, forming and growing in the center of the station. A relentless pull, dragging everything towards the abyss. A dark matter meteor is incredibly deadly.

Emagging too many meteor shields will summon a dark matt-eor. This comes with several warnings, and after awhile, warns the station that someone is trying to summon a dark matteor.

The dark matt-eor itself is not that damaging in its impact, but drops a singularity in its final resting place.

## Why It's Good For The Game

It's a new way to terrorize a round as an antagonist. Before, emagging a lot of meteor shields would basically make meteor showers the only event that can run, which is cool, but since constant meteor waves are going to destroy the station, let's also throw in the mother of all meteors!

This also adds warnings to spamming emagging meteor shields, which imo needs it. The round ends when someone spams emagged meteor shields, and since they're meteor shields nobody is going to reasonably check on them.

## Changelog
:cl:
add: The dark matt-eor
add: Summon a dark matt-eor final traitor objective
add: Dark matter singularity variant, which can't grow as big as a regular singularity but hungers for blood
code: cleaned up/sorted meteor shield code, satellite control, and more
qol: added a lot of feedback to interacting with meteor shields
balance: emagging a lot of meteor shields warns the station, but emagging enough of them summons a Dark Matt-eor.
/:cl:

tralezab committed:
file sorting, dark matt-eor

Fikou commented on code/modules/station_goals/meteor_shield.dm:
```diff
@@ -172,27 +79,61 @@
+			say("Warning. Risk of dark matter congealment entering existent ranges. Further tampering will be reported.")
+		if(EMAGGED_METEOR_SHIELD_THRESHOLD_THREE)
+			say("Warning. Further tampering has been reported.")
+			priority_announce()
```
nice announcement

tralezab replied:
oops i already fixed this locally...

tralezab committed:
missing announcement

Fikou commented on code/game/machinery/satellite/satellite_control.dm:
```diff
@@ -0,0 +1,48 @@
+			toggle(text2num(params["id"]))
+			. = TRUE
+
+/obj/machinery/computer/sat_control/proc/toggle(id)
```
mind touching this and the one below to use descriptive var names?

flowercuco commented:
no one throws spaghetti at the wall like you tralezab thank you so much

Fikou commented:
> When he designs a traitor final objective, but just puts it behind the emag instead...

just build a bsa to stop the singulo

tralezab commented:
## I'm just enjoying watching the dang thing

![image](https://user-images.githubusercontent.com/40974010/228373515-9ec46ca4-f67d-406b-a5da-6ac60ad58889.png)

Fikou commented:
> Fikou when you can't order the BSA and Meteor Sats on the same round

yeah you can when its extended

ATH1909 commented:
what's the purpose of the warnings when you can just construct 10 meteor sats in a remote location and emag them all in a row to instantly summon a singularity meteor?

ATH1909 commented:
> ATHATH i _just_ said I was considering that. BRO

okay hear me out here

instead of making this an effect for emagging too many meteor shields, you make this an actual traitor final objective that can only be given if the station has a meteor shields objective. make it, like, a device you strap onto one of the shields or something.

ATH1909 commented:
> ATHATH i _just_ said I was considering that. BRO
> 
> ![image](https://user-images.githubusercontent.com/40974010/228376738-d1c85878-bb0a-43b2-987e-1159addf6d73.png)

o

tralezab commented:
This is a neat final traitor objective, sure I'll add it as one since it's pretty adaptable into one

Kubisopplay commented:
Please let me throw that as an admin smite

tralezab committed:
see commit desc

cooldown on emagging, always announce, sparks fixed, new notify message, fixes deleting the singuloth on spawn, dark matteor is now immune to meteor shields

tralezab committed:
GPS signals emanate from corrupted meteor sats

tralezab commented:
Changes to avoid abuse and keep it a FUN feature:
- 1 minute cooldown on emagging
- always announces the dark matteor avoiding some edge cases where it collides before the announce can find it
- all hacked meteor shields give off a gps signal

tralezab committed:
now singuloth REALLY doesn't self delete

tralezab committed:
removes hard del

TheSmallBlue commented:
Should have like, a 10% chance of being named "The Big Chip" instead

LemonInTheDark commented:
Ran out from sidewalk into 6 lane road/10
Pogged up

DevinShafer commented:
> > When he designs a traitor final objective, but just puts it behind the emag instead...
> 
> just build a bsa to stop the singulo

Just, throw a bluespace bag at it, If the codes still there I believe it "shrinks" it still.
or you know, throw the clown, the clown has fun interaction

either way I am to blame for this PR to be existent and I love it

tralezab committed:
DARK MATTER SINGULOTH

tralezab committed:
variant icons

tralezab committed:
fixes dark matteor event

tralezab committed:
stop UNTICKING

tralezab committed:
the dark matter singuloth may not grow as big

but it hungers for the blood of the innocent

tralezab committed:
new sprites jesus so painful to make these

tralezab committed:
more visible nosebleeds

Pumpkinoe commented:
copyrighted pumpkin0 2023 its my idea i swear

tralezab committed:
finally fixes the dark matt-eor event

tralezab committed:
see desc

fixes bloodthirsty, the meteor triggers red alert not the announcement, the meteor will reset the alert level to what it was before if it misses, ensures that it WON'T miss (heh), lowers explosion immunity to give a small window to maxcap the singuloth to death

Singul0 commented:
> > ATHATH i _just_ said I was considering that. BRO
> 
> okay hear me out here
> 
> instead of making this an effect for emagging too many meteor shields, you make this an actual traitor final objective that can only be given if the station has a meteor shields objective. make it, like, a device you strap onto one of the shields or something.
> 
> this will satisfy the slow cook-enjoyers who just wanna bring back the meteors gamemode, still add the cool dark matter meteor to the game, and prevent people from raging over some doofus summoning a singulo 20 minutes into the round.

that assumes people actually do station goals. this will make the objective more annoying and harder to do than any other final objjectives since the traitor need to buy and build a meteorshield beforehand.

tralezab committed:
see desc again

fixes sat variable names, dark matter singulo ignores GODMODE creatures, new dark matter toy

nianjiilical commented:
who is dark matt and what have we done to incur his wrath

tralezab committed:
final objective

Watermelon914 commented on code/datums/components/singularity.dm:
```diff
@@ -335,5 +341,47 @@
+
+/// gives a little fluff warning that someone is being hunted.
+/datum/component/singularity/bloodthirsty/proc/foreboding_nosebleed(mob/living/target)
+	if(iscarbon(target))
```
you should early return here

tralezab committed:
more bloodthirsty logic, typos

singulo switches targets eventually when one is closer, typos

tralezab committed:
review

tralezab committed:
Update singularity.dm

Fikou commented on code/datums/components/singularity.dm:
```diff
@@ -335,5 +343,60 @@
+			to_chat(carbon_target, span_notice("You get a headache."))
+			return
+		head.adjustBleedStacks(5)
+		carbon_target.visible_message(span_notice("[carbon_target] gets a nosebleed."), span_notice("You get a nosebleed."))
```
```suggestion
		carbon_target.visible_message(span_notice("[carbon_target] gets a nosebleed."), span_warning("You get a nosebleed."))
```

Fikou commented on code/game/machinery/satellite/satellite_control.dm:
```diff
@@ -0,0 +1,48 @@
+		if(satellite.id == toggled_id && is_valid_z_level(get_turf(satellite), current_turf))
+			satellite.toggle()
+
+/obj/machinery/computer/sat_control/ui_data()
```
on this proc use descriptive var names too

Fikou commented on code/game/objects/items/toys.dm:
```diff
@@ -295,6 +295,19 @@
+		\"</i><br>- Chief Engineer Ship Chafer"
+	icon = 'icons/obj/engine/singularity.dmi'
+	icon_state = "dark_matter_s1"
+	item_flags = NO_PIXEL_RANDOM_DROP
```
inherited from parent

Fikou commented on code/modules/meteors/meteor_dark_matteor.dm:
```diff
@@ -0,0 +1,64 @@
+
+/obj/effect/meteor/dark_matteor/handle_stopping()
+	. = ..()
+	SSsecurity_level.set_level(previous_security_level)
```
check if its still on red here

Fikou commented on code/modules/meteors/meteor_dark_matteor.dm:
```diff
@@ -0,0 +1,64 @@
+/obj/effect/meteor/dark_matteor/Initialize(mapload, turf/target)
+	. = ..()
+	previous_security_level = SSsecurity_level.get_current_level_as_number()
+	SSsecurity_level.set_level(SEC_LEVEL_RED)
```
what happens if its already red

tralezab committed:
revieww

Loafin34 commented:
this is the first maia roentgen reference in code. very surprising.

carshalash commented:
There's seriously no warning or anything for this?

Draggeru commented:
> There's seriously no warning or anything for this?

There is unless an admin forces it.
